db/migrations/mysql/000012_create_users.up.sql

```diff
@@ -0,0 +1,242 @@
+    AuthData varchar(128) DEFAULT NULL,
+    AuthService varchar(32) DEFAULT NULL,
+    Email varchar(128) DEFAULT NULL,
+    EmailVerified tinyint(1),
+    Nickname varchar(64) DEFAULT NULL,
+    FirstName varchar(64) DEFAULT NULL,
+    LastName varchar(64) DEFAULT NULL,
+    Roles varchar(256) DEFAULT NULL,
+    AllowMarketing tinyint(1),
+    Props text,
+    NotifyProps text,
+    LastPasswordUpdate bigint(20) DEFAULT NULL,
+    LastPictureUpdate bigint(20) DEFAULT NULL,
+    FailedAttempts integer,
+    Locale varchar(5) DEFAULT NULL,
+    MfaActive tinyint(1),
+    MfaSecret varchar(128) DEFAULT NULL,
+    PRIMARY KEY (Id),
+    UNIQUE KEY Username (Username),
+    UNIQUE KEY AuthData (AuthData),
+    UNIQUE KEY Email (Email)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'Users'
+        AND table_schema = DATABASE()
+        AND column_name = 'LastActivityAt'
+    ) > 0,
+    'ALTER TABLE Users DROP COLUMN LastActivityAt;',
+    'SELECT 1'
+));
+
+PREPARE alterIfExists FROM @preparedStatement;
+EXECUTE alterIfExists;
+DEALLOCATE PREPARE alterIfExists;
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'Users'
+        AND table_schema = DATABASE()
+        AND column_name = 'LastPingAt'
+    ) > 0,
+    'ALTER TABLE Users DROP COLUMN LastPingAt;',
+    'SELECT 1'
+));
+
+PREPARE alterIfExists FROM @preparedStatement;
+EXECUTE alterIfExists;
+DEALLOCATE PREPARE alterIfExists;
+
+UPDATE Users SET Roles = 'system_user' WHERE Roles = '';
+UPDATE Users SET Roles = 'system_user system_admin' WHERE Roles = 'system_admin';
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'Users'
+        AND table_schema = DATABASE()
+        AND column_name = 'Position'
+    ) > 0,
+    'SELECT 1',
+    'ALTER TABLE Users ADD Position varchar(128);'
+));
+
+PREPARE alterIfNotExists FROM @preparedStatement;
+EXECUTE alterIfNotExists;
+DEALLOCATE PREPARE alterIfNotExists;
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'Users'
+        AND table_schema = DATABASE()
+        AND column_name = 'Timezone'
+    ) > 0,
+    'SELECT 1',
+    'ALTER TABLE Users ADD Timezone varchar(256) DEFAULT \'{"automaticTimezone":"","manualTimezone":"","useAutomaticTimezone":"true"}\';'
+));
+
+PREPARE alterIfNotExists FROM @preparedStatement;
+EXECUTE alterIfNotExists;
+DEALLOCATE PREPARE alterIfNotExists;
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'Users'
+        AND table_schema = DATABASE()
+        AND column_name = 'Position'
+    ) > 0,
+    'ALTER TABLE Users MODIFY Position varchar(128);',
```

Comment:
shouldn't we check on the length of the column before making the change? Something along the lines of: https://github.com/mattermost/mattermost-server/pull/17463/files#diff-63823a10e7e7051a9fea834d002414805988bcd8d704793385ae2b81934b79a1R19